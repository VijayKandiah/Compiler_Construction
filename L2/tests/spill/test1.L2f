(:myF
  0 1

  %myVar <- 0
  %myVar++
  %myVar += 3
  %myVar += 5
  mem rsp 0 <- %myVar

  rdi <- mem rsp 0
  call :myF2 0

  return
)

%myVar

%S
