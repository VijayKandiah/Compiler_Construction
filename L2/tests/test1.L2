(:go
  (:go
    0 0
    %v1 <- +5
    return
  )
)
