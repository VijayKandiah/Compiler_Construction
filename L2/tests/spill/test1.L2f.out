(:myF
	0 2
	%S0 <- 0
	mem rsp 8 <- %S0
	%S1 <- mem rsp 8
	%S1++
	mem rsp 8 <- %S1
	%S2 <- mem rsp 8
	%S2 += 3
	mem rsp 8 <- %S2
	%S3 <- mem rsp 8
	%S3 += 5
	mem rsp 8 <- %S3
	%S4 <- mem rsp 8
	mem rsp 0 <- %S4
	rdi <- mem rsp 0
	call :myF2 0
	return
)
